### DISCUSSION TITLE
Put all the docs build in a container image

### DISCUSSION THREAD

**Discussion by alecthegeek** [wezterm/wezterm#6201] [Ideas]
Currently the docs build uses a container image for the mkdocs tools.

However there are other dependencies (e.g. Python, gelatyz which needs Rust,...)

This creates friction for folks like me who use Wezterm, want to contribute to the docs, but may not have the dependencies installed

I propose to put all these in a single container image so that no additional software beyond a container runtime engine is needed to build the docs,

Anyone think this is a bad idea?.

Approach
========

1. Create a container image with all the tools
2. Modify the `ci/make-docs.sh` script to use the new image
3. Create a GitHub action to keep the image updated
4. Store the image on GitHub and reference from make-docs.sh

**Comment by loops** [comment]
Hey alecthegeek,

Anything that makes it easier to maintain the docs is great!  I had a bit of luck using Github
to test out docs before making a PR: 

1.  Make a personal fork of  Wezterm on Github.
2.  Goto  fork -> settings -> pages ->  Deploy from branch : set a branch, and set /docs as the directory
3.  Make local changes to the docs and then push them to that branch on Github.
4.  They show up compiled on:   https://loops.github.io/wezterm/

It's basically a mirror of the live website, with changes applied.
I have no idea what that leaves out, or if it gets close to what you
need, but thought I'd mention it.

Cheers.

**Reply by alecthegeek** [reply]
With my approach documentarians will be able to preview locally if they have a desktop container engine installed. (e.g., Rancher Desktop, Docker Desktop, Finch, or Moby for Linux workstations).

However an additional  GitHub option to preview on a branch would be a good addition for people who don't have easy CLI access (e.g. iPad users)
